src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -126,6 +130,14 @@ def _create_binary(self, binary_tgt, results_dir):
       # Dump everything into the builder's chroot.
       for tgt in source_tgts:
         dump_sources(builder, tgt, self.context.log)
+
+      # Handle locally-built python distribution dependencies.
+      built_dists = self.context.products.get_data(BuildLocalPythonDistributions.PYTHON_DISTS)
+      if built_dists:
+        req_tgts = inject_req_libs_provided_by_setup_file(self.context.build_graph,
+                                                          built_dists,
+                                                          binary_tgt.invalidation_hash(), binary_tgt) + req_tgts
```

Comment:
generally speaking, synthetic addresses would map to the path hierarchy of a given target w/ some synthetic name (e.g. `//path/to/the/dir:my_injected_dist_0`).. this maps to e.g. `//<some_sha_here>` in the root. the one thought I have is that this might eventually be bad for pantsd invalidation (changes to the root of the repo will invalidate all of these addresses in one fell swoop).

so I think at a minimum you would want to key this into the target's path namespace, e.g.:

```
os.path.join((binary_tgt.address.rel_path, binary_tgt.invalidation_hash())),
```

..it'd also be good to break out the last two params here each onto its own line for readability.

Reply:
and at that point, its probably just as safe/sane to use a static name key like:

```
':'.join((binary_tgt.address.rel_path, '{}_dist_reqs'.format(binary_tgt.address.target_name)),
```